include/exadg/aero_acoustic/user_interface/application_base.h

```diff
@@ -331,16 +331,17 @@ class ApplicationBase
     parameters.print(pcout, "List of parameters for aero-acoustic solver");
 
     field_functions = std::make_shared<FieldFunctions<dim>>();
-    set_field_functions();
   }
 
+  // has to be called after the setup of acoustic and fluid member because we want to have
+  // access to them
+  virtual void
+  set_field_functions() = 0;
+
   virtual void
   add_parameters(dealii::ParameterHandler & prm)
   {
     parameters.add_parameters(prm, "AeroAcoustic");
-
-    acoustic->add_parameters(prm);
-    fluid->add_parameters(prm);
```

Comment:
I would expect that this change makes this function https://github.com/exadg/exadg/blob/024b1224410ddd01a12ed924e107ee84838e1d14/include/exadg/aero_acoustic/solver.h#L37 incorrect.

Reply:
it seems as if you only want to change the private member function `parse_parameters()`. If so, one would have to change the implementation of that private member function. For clarity, a comment should be added that this private member function only parses the parameters of the coupling problem, not the parameters of the single-field solvers.